karabo_bridge/client.py

```diff
@@ -20,6 +19,9 @@
 __all__ = ['Client']
 
 
+CONTEXT = zmq.Context()
```

Comment:
Is there a specific reason for clients to share a context? I'd have thought separate contexts were neater: it avoids things happening at import time, it avoids unexpected teardown operations when the Python interpreter is cleaning up, and it leaves multiple clients fewer ways to interfere with each other (e.g. I think with a shared context, receiving a large message on one socket could hold up receiving smaller messages on other sockets).

Reply:
I thought I've read a while back that best practice was to have a single context per process, but researching again now, results are not so categorical... I'll change that back

Reply:
If I recall correctly, each context creates a (C level) thread to read from sockets. So you probably don't want 1000 separate contexts around. But if there are multiple logical 'programs within a program', it might make sense for them to have independent contexts.